### PR TITLE
README: fix everything

### PR DESCRIPTION
**Pull request by slimsag**
My gosh, I hadn't looked at this README in a while but it has become a disaster. 😄When we need "quick links" at the top of the README above the actual explanation for what the thing does -- I think that's the point at which restructuring documentation starts to make a lot of sense :)

Please compare:

- Before: https://github.com/sourcegraph/src-cli/blob/0001a822905f3e292b18c45987bf479f6a92019d/README.md
- After: https://github.com/sourcegraph/src-cli/blob/sg/readme/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,186 +1,117 @@
...
+ - `src campaigns` - manages [campaigns](https://docs.sourcegraph.com/user/campaigns)
+ - `src repos` - manage repositories
+ - `src users` - manage users
+ - `src orgs` - manages organization
+ - `src config` - manage global, org, and user settings
+ - `src extsvc` - manage external services (repository configuration)
+ - `src extensions` - manage extensions
  - `lsif` - manages LSIF data
- - `version` - display and compare the src-cli version against the recommended version of the configured Sourcegraph instance
+ - `version` - check for updates
```
Comment (efritz) [inline README.md:105]: src lsif/src version?

Comment (slimsag) [inline README.md:105]:
```suggestion
 - `src version` - check current version and guaranteed-compatible version for connected Sourcegraph instance
```

File: README.md [hunk 2]
```diff
@@ -1,186 +1,117 @@
...
+ - `src actions` - run [campaign actions](https://docs.sourcegraph.com/user/campaigns/actions) to generate patch sets
+ - `src api` - run Sourcegraph GraphQL API requests
+ - `src campaigns` - manages [campaigns](https://docs.sourcegraph.com/user/campaigns)
+ - `src repos` - manage repositories
+ - `src users` - manage users
+ - `src orgs` - manages organization
+ - `src config` - manage global, org, and user settings
+ - `src extsvc` - manage external services (repository configuration)
+ - `src extensions` - manage extensions
  - `lsif` - manages LSIF data
```
Comment (efritz) [inline README.md:104]:
```suggestion
 - `src lsif` - manages LSIF data
```

**Comment by slimsag** [comment]
Gonna merge, feel free to send PRs or leave comments!

**Pull request merged**